src/TLuaInterpreter.cpp

```diff
@@ -11466,25 +11495,176 @@ bool TLuaInterpreter::callEventHandler(const QString& function, const TEvent& pE
 
     for (int i = 0; i < pE.mArgumentList.size(); i++) {
         switch (pE.mArgumentTypeList.at(i)) {
-            case ARGUMENT_TYPE_NUMBER:
-                lua_pushnumber(L, pE.mArgumentList.at(i).toDouble());
-                break;
-            case ARGUMENT_TYPE_STRING:
-                lua_pushstring(L, pE.mArgumentList.at(i).toUtf8().constData());
-                break;
-            case ARGUMENT_TYPE_BOOLEAN:
-                lua_pushboolean(L, pE.mArgumentList.at(i).toInt());
-                break;
-            case ARGUMENT_TYPE_NIL:
-                lua_pushnil(L);
-                break;
-            default:
-                qWarning(R"(TLuaInterpreter::callEventHandler("%s", TEvent) ERROR: Unhandled ARGUMENT_TYPE: %i encountered in argument %i.)", function.toUtf8().constData(), pE.mArgumentTypeList.at(i), i);
-                lua_pushnil(L);
+        case ARGUMENT_TYPE_NUMBER:
+            lua_pushnumber(L, pE.mArgumentList.at(i).toDouble());
+            break;
+        case ARGUMENT_TYPE_STRING:
+            lua_pushstring(L, pE.mArgumentList.at(i).toUtf8().constData());
+            break;
+        case ARGUMENT_TYPE_BOOLEAN:
+            lua_pushboolean(L, pE.mArgumentList.at(i).toInt());
+            break;
+        case ARGUMENT_TYPE_NIL:
+            lua_pushnil(L);
+            break;
+        default:
+            qWarning(R"(TLuaInterpreter::callEventHandler("%s", TEvent) ERROR: Unhandled ARGUMENT_TYPE: %i encountered in argument %i.)", function.toUtf8().constData(), pE.mArgumentTypeList.at(i), i);
+            lua_pushnil(L);
         }
     }
 
-    error = lua_pcall(L, pE.mArgumentList.size(), LUA_MULTRET, 0);
+    if (qE != 0) {
+        // Create Lua table with QEvent data if needed
+        switch (qE->type()) {
+        // This means the default argument value was used, so ignore
+        case (QEvent::None):
+            error = lua_pcall(L, pE.mArgumentList.size(), LUA_MULTRET, 0);
+            break;
+        // These are all QMouseEvents
+        case (QEvent::MouseButtonPress):
+        case (QEvent::MouseButtonDblClick):
+        case (QEvent::MouseButtonRelease):
+        case (QEvent::MouseMove): {
+            auto qME = static_cast<const QMouseEvent*>(qE);
+            lua_newtable(L);
+
+            // push button()
+            lua_pushstring(L, QStringLiteral("button").toUtf8().constData());
+            lua_pushstring(L, mMouseButtons.value(qME->button()).toUtf8().constData());
+            lua_settable(L, -3);
```

Comment:
The triple combination can be simplified a bit and made more readable (which I prefer):
```cpp
lua_pushstring(L, mMouseButtons.value(qME->button()).toUtf8().constData());
lua_setfield(L, -2, QStringLiteral("button").toUtf8().constData()); // Use -2 as we only pushed one value to the stack
```
I'm also not sure if we really need the `QStringLiteral` here or if we could use a simple string literal as the function accepts a `const char*` anyways. But what do I know? @vadi2 and @SlySven should have a say about that.

Reply:
Yeah, good catch, this case `lua_setfield(L, -2, "button");` is enough because we're working with Lua's C function. We using `QStringLiteral` when we want a `QString()` object and that's for working with Qt.